joint_trajectory_controller/test/test_trajectory.cpp

```diff
@@ -280,3 +280,353 @@ TEST(TestTrajectory, interpolation_pos_vel_accel)
     EXPECT_NEAR(end_state.accelerations[0], expected_state.accelerations[0], EPS);
   }
 }
+
+TEST(TestTrajectory, sample_trajectory_velocity_with_interpolation)
+{
+  auto full_msg = std::make_shared<trajectory_msgs::msg::JointTrajectory>();
+  full_msg->header.stamp = rclcpp::Time(0);
+
+  // definitions
+  double time_first_seg = 1.0;
+  double time_second_seg = 2.0;
+  double time_third_seg = 3.0;
+  double velocity_first_seg = 1.0;
+  double velocity_second_seg = 2.0;
+  double velocity_third_seg = 1.0;
+
+  trajectory_msgs::msg::JointTrajectoryPoint p1;
+  p1.velocities.push_back(velocity_first_seg);
+  p1.time_from_start = rclcpp::Duration::from_seconds(time_first_seg);
+  full_msg->points.push_back(p1);
+
+  trajectory_msgs::msg::JointTrajectoryPoint p2;
+  p2.velocities.push_back(velocity_second_seg);
+  p2.time_from_start = rclcpp::Duration::from_seconds(time_second_seg);
+  full_msg->points.push_back(p2);
+
+  trajectory_msgs::msg::JointTrajectoryPoint p3;
+  p3.velocities.push_back(velocity_third_seg);
+  p3.time_from_start = rclcpp::Duration::from_seconds(time_third_seg);
+  full_msg->points.push_back(p3);
+
+  trajectory_msgs::msg::JointTrajectoryPoint point_before_msg;
+  point_before_msg.time_from_start = rclcpp::Duration::from_seconds(0.0);
+  point_before_msg.positions.push_back(0.0);
+  point_before_msg.velocities.push_back(0.0);
+
+  // set current state before trajectory msg was sent
+  const rclcpp::Time time_now = rclcpp::Clock().now();
+  auto traj = joint_trajectory_controller::Trajectory(time_now, point_before_msg, full_msg);
+
+  trajectory_msgs::msg::JointTrajectoryPoint expected_state;
+  joint_trajectory_controller::TrajectoryPointConstIter start, end;
+
+  // sample at trajectory starting time
+  {
+    traj.sample(time_now, expected_state, start, end);
+    EXPECT_EQ(traj.begin(), start);
+    EXPECT_EQ(traj.begin(), end);
+    EXPECT_NEAR(point_before_msg.positions[0], expected_state.positions[0], EPS);
+    EXPECT_NEAR(point_before_msg.velocities[0], expected_state.velocities[0], EPS);
+    // is 0 because point_before_msg does not have velocity defined
+    EXPECT_NEAR(1.0, expected_state.accelerations[0], EPS);
```

Comment:
Could you document in a comment why you expect 1.0 for the sake of future test extensions?